src/clients/circleci/projects.ts

```diff
@@ -0,0 +1,29 @@
+import { Project } from '../schemas.js';
+import { HTTPClient } from './httpClient.js';
+
+export class ProjectsAPI {
+  protected client: HTTPClient;
+
+  constructor(httpClient: HTTPClient) {
+    this.client = httpClient;
+  }
+
+  /**
+   * Get all workflows for a pipeline with pagination support
+   * @param params Configuration parameters
+   * @param params.projectSlug The project slug
+   * @returns Flaky test details
+   * @throws Error if timeout or max pages reached
+   */
```

Comment:
Need to update the doc string here. Maybe something like this:
```suggestion
  /**
   * Get project info by slug
   * @param projectSlug The project slug
   * @returns The project info
   * @throws Error if the request fails
   */
```